utils/python/CIME/hist_utils.py

```diff
@@ -282,9 +301,23 @@ def generate_baseline(case, baseline_dir=None):
             if os.path.exists(baseline):
                 os.remove(baseline)
 
+            #special care for multi-instance cases,
+            #only keep first instance and
```

Comment:
Why do you go to the trouble of only keeping the first instance? I know that, in general use currently, the _0001 and _0002 instances are identical. But it would be good if we started having multi-instance tests where the instances actually differed in some way. So it seems like it would be good to just go ahead and save (and then compare) all instances - unless there's some problem with doing that that I'm not seeing.